plutus-tx/src/PlutusTx/Blueprint/Parameter.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE KindSignatures     #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE RecordWildCards    #-}
+
+module PlutusTx.Blueprint.Parameter where
+
+import Prelude
+
+import Data.Aeson (ToJSON (..), (.=))
+import Data.Aeson qualified as Aeson
+import Data.Aeson.Extra (optionalField, requiredField)
+import Data.Aeson.KeyMap qualified as KeyMap
+import Data.Function ((&))
+import Data.Kind (Type)
+import Data.Set (Set)
+import Data.Set qualified as Set
+import Data.Text (Text)
+import PlutusTx.Blueprint.Purpose (Purpose)
+import PlutusTx.Blueprint.Schema (Schema)
+
+{- | Blueprint that defines validator's compile-time parameter.
+
+  The 'referencedTypes' phantom type parameter is used to track the types used in the contract
+  making sure their schemas are included in the blueprint and that they are referenced
+  in a type-safe way.
+-}
+data ParameterBlueprint (referencedTypes :: [Type]) = MkParameterBlueprint
+  { parameterTitle       :: Maybe Text
+  -- ^ A short and descriptive name for the parameter.
+  , parameterDescription :: Maybe Text
+  -- ^ An informative description of the parameter.
+  , parameterPurpose     :: Set Purpose
```

Comment:
Why is this a set?

Reply:
To rule out an invalid state of having same purpose more than one time, e.g.
```haskell
parameterPurpose = [Purpose.Spend, Purpose.Spend, Purpose.Mint]
```

Reply:
What I meant to ask is why does a parameter have multiple purposes? Isn't it a oneOf? In fact I don't understand why a parameter should have any purpose at all, I thought purpose is something for a script, not for an individual parameter.